src/test/java/io/cdap/plugin/couchbase/etl/CouchbaseSourceETLTest.java

```diff
@@ -156,20 +157,77 @@ public void testSource() throws Exception {
       Assert.assertNotNull(actualCreatedAt);
       JsonDocument expected = getTestDocumentByCreationTime(actualCreatedAt);
       Assert.assertNotNull(expected);
+      assertNumericEquals(expected, actual);
+      assertMapEquals("object_map", expected, actual);
+      assertRecordEquals("object", expected, actual);
+      assertEquals(expected, actual);
+    }
+  }
+
+  @Test
+  public void testSourceSingleSplit() throws Exception {
+    Map<String, String> properties = sourceProperties("*", SCHEMA, 1);
+    List<StructuredRecord> records = getPipelineResults(properties);
+    Assert.assertEquals(TEST_DOCUMENTS.size(), records.size());
+    for (StructuredRecord actual : records) {
+      Long actualCreatedAt = actual.get("created");
+      Assert.assertNotNull(actualCreatedAt);
+      JsonDocument expected = getTestDocumentByCreationTime(actualCreatedAt);
+      Assert.assertNotNull(expected);
+      assertNumericEquals(expected, actual);
+      assertMapEquals("object_map", expected, actual);
+      assertRecordEquals("object", expected, actual);
+      assertEquals(expected, actual);
+    }
+  }
+
+  @Test
+  public void testSourceTwoSplits() throws Exception {
+    Map<String, String> properties = sourceProperties("*", SCHEMA, 2);
+    List<StructuredRecord> records = getPipelineResults(properties);
+    Assert.assertEquals(TEST_DOCUMENTS.size(), records.size());
+    for (StructuredRecord actual : records) {
+      Long actualCreatedAt = actual.get("created");
+      Assert.assertNotNull(actualCreatedAt);
+      JsonDocument expected = getTestDocumentByCreationTime(actualCreatedAt);
+      Assert.assertNotNull(expected);
+      assertNumericEquals(expected, actual);
+      assertMapEquals("object_map", expected, actual);
+      assertRecordEquals("object", expected, actual);
+      assertEquals(expected, actual);
+    }
+  }
+
+  @Test
+  public void testSourceNumSplitsGreaterThanDocumentsNumber() throws Exception {
+    Map<String, String> properties = sourceProperties("*", SCHEMA, 20);
+    List<StructuredRecord> records = getPipelineResults(properties);
+    Assert.assertEquals(TEST_DOCUMENTS.size(), records.size());
+    for (StructuredRecord actual : records) {
```

Comment:
it seems there are too many code duplication in tests

Reply:
Reviewed for duplication in the `CouchbaseSourceETLTest`, `CouchbaseSinkETLTest` and config tests.